lms/djangoapps/courseware/views/views.py

```diff
@@ -726,6 +726,12 @@ def _progress(request, course_key, student_id):
 
     # checking certificate generation configuration
     enrollment_mode, is_active = CourseEnrollment.enrollment_mode_for_user(student, course_key)
+
+    # We need to have id verified to show generate cert button
+    # in the case where courseMode is of Verified variety.
+    is_id_verified = SoftwareSecurePhotoVerification.user_is_verified(student) or \
+        enrollment_mode and not CourseMode.has_verified_mode(enrollment_mode)
```

Comment:
@schenedx , I think the logic written here is something like: if the user isn't verified, then check to make sure that they're enrolled in the course and the course does not have a verified mode. This check would return False if:
1) The user was enrolled in the course in the "honor" track, but the course also had a "verified track"
2) The user was enrolled in the "verified" track, but wasn't photoverified

I think we want logic more like:
```python
SoftwareSecurePhotoVerification.user_is_verified(student) or enrollment_mode == CouseMode.HONOR
```